src/charm.py

```diff
@@ -571,6 +563,72 @@ def _gnb_subnet_config_is_valid(self) -> bool:
     def _get_gnb_subnet_config(self) -> Optional[str]:
         return self.model.config.get("gnb-subnet")
 
+    def _get_access_nad_config(self) -> Dict[Any, Any]:
+        return {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "routes": [
+                    {
+                        "dst": self._get_gnb_subnet_config(),
+                        "gw": self._get_access_network_gateway_ip_config(),
+                    },
+                ],
+                "addresses": [
+                    {
+                        "address": self._get_access_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+
+    def _get_core_nad_config(self) -> Dict[Any, Any]:
+        return {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._get_core_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+
+    def _get_core_interface_mtu_config(self) -> Optional[int]:
+        return self.model.config.get("core-interface-mtu-size")  # type: ignore
```

Comment:
Does this actually return `int`? I thought Juju configs are strings by default, aren't they?

Reply:
Yes, it is returning int.

Reply:
It does not return `int`, but `Optional[str]`.

Please remove the `# type: ignore` and update the type hint for the return type of the method.

Reply:
It does not seems correct to me. We are indicating type int in config.yaml, however setting it Optional[str] just to fix the type problem here but creates another typing issue in the validation of value.

Reply:
Returned as str, then str -> int validation is done in validation.